values/decoding/fs_string_values.go

```diff
@@ -60,14 +61,38 @@ func (v *fsBasedStringValues) Iter() (iterator.ForwardStringIterator, error) {
 	return newStringIteratorFromMeta(v.metaProto, v.encodedValues, v.dictArr, v.encodedDictBytes)
 }
 
-// TODO(xichen): Filter implementation should take advantage of the metadata
-// to do more intelligent filtering, e.g., checking if the value is within the
-// value range, intelligently look up filter values and bail early if not found,
-// perform filtering directly against the index to avoid string comparisons.
+// TODO(xichen): Filter implementation should intelligently look up filter values and bail early if not found
```

Comment:
Perhaps handle the rest of the TODOs as well?

Reply:
The rest of the TODOs are more involved. Let's save them for a future PR and get these more minor changes in now.